packages/console-extension/src/index.ts

```diff
@@ -374,26 +374,42 @@ async function activateConsole(
   const pluginId = '@jupyterlab/console-extension:tracker';
   let interactionMode: string;
   let promptCellConfig: JSONObject;
-  async function updateSettings() {
+
+  /**
+   * Update settings for one console or all consoles.
+   *
+   * @param panel Optional - single console to update.
+   */
+  async function updateSettings(panel?: ConsolePanel) {
     interactionMode = (await settingRegistry.get(pluginId, 'interactionMode'))
       .composite as string;
     promptCellConfig = (await settingRegistry.get(pluginId, 'promptCellConfig'))
       .composite as JSONObject;
-    tracker.forEach(widget => {
+
+    const setWidgetOptions = (widget: ConsolePanel) => {
       widget.console.node.dataset.jpInteractionMode = interactionMode;
       setOption(widget.console.promptCell?.editor, promptCellConfig);
-    });
+    };
+
+    if (panel) {
+      setWidgetOptions(panel);
+    } else {
+      tracker.forEach(widget => {
```

Comment:
Small suggestion, take it or leave it: can we just use the function directly here and simplify this to `tracker.forEach(setWidgetOptions)`?